cmd/device/list/list.go

```diff
@@ -29,45 +27,41 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
+const deviceTempl = "Device ID\tDevice Name\tOperating State\tAdmin State\tDevice Service\tDevice Profile\n" +
+	"{{range .}}" +
+	"{{.Id}}\t{{.Name}}\t{{.OperatingState}}\t{{.AdminState}}\t{{.Service.Name}}\t{{.Profile.Name}}\n" +
+	"{{end}}"
 
 // NewCommand returns the list device command
 func NewCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "list",
 		Short: "A list of all device services",
 		Long:  `Return all device services sorted by id.`,
-		RunE: func(cmd *cobra.Command, args []string) (err error) {
-			url := config.Conf.Clients["Metadata"].Url() + clients.ApiDeviceRoute
-			mdc := metadata.NewDeviceClient(
-				local.New(url),
-			)
+		RunE: listHandler,
+	}
+	return cmd
+}
 
-			devices, err := mdc.Devices(context.Background())
-			if err != nil {
-				return
-			}
+func listHandler(cmd *cobra.Command, args []string) (err error) {
+	url := config.Conf.Clients["Metadata"].Url() + clients.ApiDeviceRoute
+	mdc := metadata.NewDeviceClient(
+		local.New(url),
+	)
 
-			pw := viper.Get("writer").(io.Writer)
-			w := new(tabwriter.Writer)
-			w.Init(pw, 0, 8, 3, ' ', 0)
-			//TODO should we always check for err retuinred from fmt.Fprintf ?
-			_, err = fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t\n", "Device ID", "Device Name", "Operating State", "Device Service", "Device Profile")
-			if err != nil {
-				return
-			}
-			for _, device := range devices {
-				fmt.Fprintf(w, "%s\t%s\t%v\t%s\t%s\t\n",
-					device.Id,
-					device.Name,
-					device.OperatingState,
-					device.Service.Name,
-					device.Profile.Name,
-				)
-			}
-			//TODO we do not constantly check for errors returned by w.Flush(). SHould we do it in the entire project ?
-			err = w.Flush()
-			return
-		},
+	devices, err := mdc.Devices(context.Background())
+	if err != nil {
+		return
 	}
-	return cmd
+
+	formatter := createFormatter()
+	err = formatter.Write(devices)
+	return
+}
+
+func createFormatter() formatters.FormatWriter {
```

Comment:
You have this method defined in multiple places, consider refactoring, taking in arguments such as, string, template, and supported methods. Possibly defined in the same file as you define the empty formatter and interface. And if you did that do you still need an interface?

Reply:
yes, I intentionally named all these functions  with the same name.  
Yes, I could create one function that accepts these arguments in a single place and use it elsewhere... but i was thinking that in the future these functions could deffer and I was not sure how to approach it now ,so I intended to address it in follow-up PR when the idea is clear to me. 
But ok, let do it now, and if the these function start to defer ... we will address them